error compiling scipion with gcc 6.1.1

Just updated to the latest gcc:
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/6.1.1/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++ --enable-shared --enable-threads=posix --enable-libmpx --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --disable-multilib --disable-werror --enable-checking=release
Thread model: posix
gcc version 6.1.1 20160501 (GCC)
```
and got an error when compiling ``g++ -o software/em/xmipp/libraries/data/xvsmooth.os -c -g -fPIC -Isoftware/em/xmipp/external -Isoftware/em/xmipp/libraries -Isoftware/include software/em/xmipp/libraries/data/xvsmooth.cpp``

```
software/em/xmipp/libraries/data/xvsmooth.cpp: In function ‘byte* Smooth(byte*, size_t, size_t, size_t, size_t)’:
software/em/xmipp/libraries/data/xvsmooth.cpp:171:16: error: call of overloaded ‘abs(size_t&)’ is ambiguous
    apy = abs(py);
                ^
In file included from /usr/include/c++/6.1.1/cstdlib:75:0,
                 from /usr/include/c++/6.1.1/stdlib.h:36,
                 from software/em/xmipp/libraries/data/xvsmooth.cpp:77:
/usr/include/stdlib.h:774:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
In file included from /usr/include/c++/6.1.1/stdlib.h:36:0,
                 from software/em/xmipp/libraries/data/xvsmooth.cpp:77:
/usr/include/c++/6.1.1/cstdlib:185:3: note: candidate: __int128 std::abs(__int128)
   abs(__GLIBCXX_TYPE_INT_N_0 __x) { return __x >= 0 ? __x : -__x; }
   ^~~
/usr/include/c++/6.1.1/cstdlib:180:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/6.1.1/cstdlib:172:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
software/em/xmipp/libraries/data/xvsmooth.cpp:204:18: error: call of overloaded ‘abs(size_t&)’ is ambiguous
      apx = abs(px);
                  ^
In file included from /usr/include/c++/6.1.1/cstdlib:75:0,
                 from /usr/include/c++/6.1.1/stdlib.h:36,
                 from software/em/xmipp/libraries/data/xvsmooth.cpp:77:
/usr/include/stdlib.h:774:12: note: candidate: int abs(int)
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
In file included from /usr/include/c++/6.1.1/stdlib.h:36:0,
                 from software/em/xmipp/libraries/data/xvsmooth.cpp:77:
/usr/include/c++/6.1.1/cstdlib:185:3: note: candidate: __int128 std::abs(__int128)
   abs(__GLIBCXX_TYPE_INT_N_0 __x) { return __x >= 0 ? __x : -__x; }
   ^~~
/usr/include/c++/6.1.1/cstdlib:180:3: note: candidate: long long int std::abs(long long int)
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/6.1.1/cstdlib:172:3: note: candidate: long int std::abs(long int)
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
```

Quick googling [showed](https://gcc.gnu.org/gcc-6/porting_to.html) that:

**Call of overloaded 'abs(unsigned int&)' is ambiguous**

The additional overloads can cause the compiler to reject invalid code that was accepted before. An example of such code is the below:

```
#include <stdlib.h>
int
foo (unsigned x)
{
  return abs (x);
}
```
Since calling abs() on an unsigned value doesn't make sense, this code will become explicitly invalid as per discussion in the LWG.

**Suggested fix:**

```
--- a/software/em/xmipp/libraries/data/xvsmooth.cpp
+++ b/software/em/xmipp/libraries/data/xvsmooth.cpp
@@ -168,7 +168,7 @@ byte *Smooth(byte *picSrc8, size_t swide, size_t shigh, size_t dwide, size_t dhi
                                if (y1 > shigh - 1)
                                        y1 = shigh - 1;
                        }
-                       apy = abs(py);
+                       apy = abs((int) py);
                        apy_100 = 100 - apy;
 
                        cyOff = (size_t) cy * swide; /* current line */
@@ -201,7 +201,7 @@ byte *Smooth(byte *picSrc8, size_t swide, size_t shigh, size_t dwide, size_t dhi
                                        *ptrPicSmooth++ = cD;
                                else {
                                        /* compute weighting factors */
-                                       apx = abs(px);
+                                       apx = abs((int) px);
                                        apx_100 = 100 - apx;
                                        pA = apx * apy;
                                        pB = apy * apx_100;
```

--- a/software/em/xmipp/libraries/data/xvsmooth.cpp
+++ b/software/em/xmipp/libraries/data/xvsmooth.cpp
@@ -168,7 +168,7 @@ byte *Smooth(byte *picSrc8, size_t swide, size_t shigh, size_t dwide, size_t dhi
 				if (y1 > shigh - 1)
 					y1 = shigh - 1;
 			}
-			apy = abs(py);
+			apy = abs((int) py);
 			apy_100 = 100 - apy;
 
 			cyOff = (size_t) cy * swide; /* current line */
@@ -201,7 +201,7 @@ byte *Smooth(byte *picSrc8, size_t swide, size_t shigh, size_t dwide, size_t dhi
 					*ptrPicSmooth++ = cD;
 				else {
 					/* compute weighting factors */
-					apx = abs(px);
+					apx = abs((int) px);
 					apx_100 = 100 - apx;
 					pA = apx * apy;
 					pB = apy * apx_100;

